buildSrc/src/main/java/com/google/firebase/gradle/plugins/semver/ApiDiffer.kt

```diff
@@ -0,0 +1,577 @@
+package com.google.firebase.gradle.plugins.semver
+
+import com.google.firebase.gradle.plugins.GmavenHelper
+import com.google.firebase.gradle.plugins.UnzipAar
+import java.io.BufferedInputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.jar.JarEntry
+import java.util.jar.JarInputStream
+import org.gradle.api.DefaultTask
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.Type
+import org.objectweb.asm.tree.ClassNode
+import org.objectweb.asm.tree.FieldNode
+import org.objectweb.asm.tree.MethodNode
+
+abstract class ApiDiffer : DefaultTask() {
+  @get:Input abstract val aarPath: Property<File>
+
+  @get:Input abstract val artifactId: Property<String>
+
+  @get:Input abstract val groupId: Property<String>
+
+  private val CLASS_EXTENSION = ".class"
+
+  @TaskAction
+  fun run() {
+    val gMavenHelper = GmavenHelper(groupId.get(), artifactId.get())
+    val lastAarPath = gMavenHelper.getAarFileForVersion(gMavenHelper.getLatestReleasedVersion())
+    val previousAarPath: String = aarPath.get().absolutePath.replace(".aar", "/old.aar")
+    try {
+      BufferedInputStream(URL(lastAarPath).openStream()).use { `in` ->
+        FileOutputStream(previousAarPath).use { fileOutputStream ->
+          val dataBuffer = ByteArray(1024)
+          var bytesRead: Int
+          while (`in`.read(dataBuffer, 0, 1024).also { bytesRead = it } != -1) {
+            fileOutputStream.write(dataBuffer, 0, bytesRead)
+          }
+        }
+      }
+    } catch (e: IOException) {
+      // handle exception
+    }
```

Comment:
I think you could make this app take both files as input, and make the plugin make the copy using the copy task 

https://docs.gradle.org/current/dsl/org.gradle.api.tasks.Copy.html

Reply:
hmm since one of them is a url we would not be able to use the copy task